docs/modules/data-structures/pages/working-with-external-data.adoc

```diff
@@ -208,25 +284,628 @@ hazelcast:
+</hazelcast>
+----
+--
+YAML:: 
++ 
+--
+[source,yaml]
+----
+hazelcast:
+  map:
+    default:
+      map-store:
+        enabled: true
+        class-name: com.hazelcast.examples.DummyStore
+        initial-mode: LAZY
+----
+--
+Java:: 
++ 
+--
+[source,java]
+----
+MapConfig mapConfig = new MapConfig("default");
+
+MapStoreConfig mapStoreConfig = new MapStoreConfig();
+mapStoreConfig.setClassName("com.hazelcast.examples.DummyStore");
+mapStoreConfig.setInitialLoadMode(MapStoreConfig.InitialLoadMode.LAZY)
+
+mapConfig.setMapStoreConfig(mapStoreConfig);
+----
+--
+====
+
+|[[external-data-store-ref]]`external-data-store-ref`
+|The name of the configured external
+data store to use to set up a mapping. Use this property when you have an <<existing-connection, existing connection to an external data store>>.
+
+a|`''` (empty)
+|
+
+[tabs] 
+==== 
+XML:: 
++ 
+--
+[source,xml]
+----
+<hazelcast>
+  <map name="default">
+    <map-store enabled="true">
+      <class-name>com.hazelcast.mapstore.GenericMapStore</class-name>
+      <properties>
+          <property name="external-data-store-ref">my-mysql-database</property>
+      </properties>
+    </map-store>
+</hazelcast>
+----
+--
+YAML:: 
++ 
+--
+[source,yaml]
+----
+hazelcast:
+  map:
+    default:
+      map-store:
+        enabled: true
+        class-name: com.hazelcast.mapstore.GenericMapStore
+        properties:
+            external-data-store-ref: my-mysql-database
+----
+--
+Java:: 
++ 
+--
+[source,java]
+----
+MapConfig mapConfig = new MapConfig("default");
+
+MapStoreConfig mapStoreConfig = new MapStoreConfig();
+mapStoreConfig.setClassName(GenericMapStore.class.getName());
+mapStoreConfig.setProperty("external-data-store-ref", "my-mysql-database");
+
+mapConfig.setMapStoreConfig(mapStoreConfig);
+----
+--
+====
+
+|[[table-name]]`table-name`
+|Name of the table to read from.
+
+a|The name of the map.
+|
+
+[tabs] 
+==== 
+XML:: 
++ 
+--
+[source,xml]
+----
+<hazelcast>
+  <map name="default">
+    <map-store enabled="true">
+      <class-name>com.hazelcast.mapstore.GenericMapStore</class-name>
+      <properties>
+          <property name="external-data-store-ref">my-mysql-database</property>
+          <property name="table-name">test</property>
+      </properties>
+    </map-store>
+</hazelcast>
+----
+--
+YAML:: 
++ 
+--
+[source,yaml]
+----
+hazelcast:
+  map:
+    default:
+      map-store:
+        enabled: true
+        class-name: com.hazelcast.mapstore.GenericMapStore
+        properties:
+            external-data-store-ref: my-mysql-database
+            table-name: test
+----
+--
+Java:: 
++ 
+--
+[source,java]
+----
+MapConfig mapConfig = new MapConfig("default");
+
+MapStoreConfig mapStoreConfig = new MapStoreConfig();
+mapStoreConfig.setClassName(GenericMapStore.class.getName());
+mapStoreConfig.setProperty("external-data-store-ref", "my-mysql-database");
+mapStoreConfig.setProperty("table-name", "test");
+
+mapConfig.setMapStoreConfig(mapStoreConfig);
+----
+--
```

Comment:
This again doesn't fit well, we have configuration options under the `map-store` element. They apply to all implementations. Then there is couple of properties that apply only to `GenericMapStore` and go into the `properties` section.

These should be separate.